### PR TITLE
add keyboard mw80

### PR DESCRIPTION
**Pull request by TW59420**
## Description
MWStudio is a custom keyboard enthusiast
Upload qmk firmware of mw80 PCB

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Dupe of [#17558](https://github.com/qmk/qmk_firmware/pull/17558).

**Comment by tzarc** [comment]
Reopening as the other was closed and I didn't see it.

**Comment by tzarc** [comment]
In the future, please refrain from opening multiple PRs -- the original could have been reused and ensured context was maintained between reviews.

**Comment by TW59420** [comment]
Sorry, there were some issues with the previous pr, so pulled the new branch

File: keyboards/mwstudio/mw80/mw80.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* Copyright 2022 TW59420 <https://github.com/TW59420>
```
Comment (waffle87) [inline keyboards/mwstudio/mw80/mw80.h:1]: This file should be removed.

File: keyboards/mwstudio/mw80/info.json [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    },
+    "usb": {
+        "pid": "0x7BA1",
+        "vid": "0x8001",
+        "device_version": "1.0.0"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+				{ "matrix": [0, 0], "x": 0, "y": 0 },
```
Comment (waffle87) [inline keyboards/mwstudio/mw80/info.json:48]:
```suggestion
                { "matrix": [0, 0], "x": 0, "y": 0 },
```

File: keyboards/mwstudio/mw80/info.json [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        "caps_lock": "A0"
+    },
+    "rgblight": {
+        "led_count": 16,
+        "pin": "B8",
+        "hue_steps": 10,
+        "saturation_steps": 17,
+        "brightness_steps": 17,
+        "animations": {
+            "all": true
```
Comment (waffle87) [inline keyboards/mwstudio/mw80/info.json:32]:
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

File: keyboards/mwstudio/mw80/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+Flashing example for this keyboard:
+
+    make mwstudio/mw80:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+* **Physical reset button**: Short press the button on the back of the PCB to enter the Bootloader and flash the firmware
+* **Keycode in layout**: Press the key mapped to RESET if it is available
```
Comment (waffle87) [inline keyboards/mwstudio/mw80/readme.md:22]:
```suggestion
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/mwstudio/mw80/mw80.c [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mw80.h"
+
```
Comment (waffle87) [inline keyboards/mwstudio/mw80/mw80.c:18]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**